yarn-project/sequencer-client/src/block_builder/circuit_block_builder.test.ts

```diff
@@ -302,8 +302,10 @@ describe('sequencer/circuit_block_builder', () => {
       );
       const txs = [tx, await makeEmptyProcessedTx(), await makeEmptyProcessedTx(), await makeEmptyProcessedTx()];
 
-      const [l2Block] = await builder.buildL2Block(blockNumber, txs, mockL1ToL2Messages);
+      const [l2Block, , rootRollupPublicInputs] = await builder.buildL2Block(blockNumber, txs, mockL1ToL2Messages);
+      
       expect(l2Block.number).toEqual(blockNumber);
+      expect(l2Block.getCalldataHash()).toEqual(rootRollupPublicInputs.sha256CalldataHash());
```

Comment:
If this check is the only reason why we're exposing the root rollup simulation output from the block builder, then I'd make it a responsibility of the block builder to make this check instead, and revert the change to its interface.

Reply:
@cheethas and I spoke about it. Main annoyance with that it broke some tests, but probably better to fix those mocks anyway.